scanner/config/config.go

```diff
@@ -185,9 +186,18 @@ func (c *MatcherConfig) validate() error {
 			return fmt.Errorf("indexer_addr: failed to parse address: %w", err)
 		}
 	}
+
+	if c.VulnerabilitiesURL == "" {
+		return errors.New("vulnerabilities_url: cannot be empty")
+	}
 	if _, err := url.Parse(c.VulnerabilitiesURL); err != nil {
-		return fmt.Errorf("vulnerabilities_url: %w", err)
+		return fmt.Errorf("vulnerabilities_url: invalid URL: %w", err)
+	}
+	v := "dev"
+	if buildinfo.ReleaseBuild {
+		v = version.Version
 	}
+	c.VulnerabilitiesURL = strings.ReplaceAll(c.VulnerabilitiesURL, "ROX_VERSION", v)
```

Comment:
I think it would make it easier to document, expand, and maintain this if we use https://pkg.go.dev/text/template

The `data` object could be just `struct {Version string}` for now, and we could only parse the URL; but moving forward, if more complex configuration patterns arise, we wouldn't have to keep backward compatibility with a random replacement of `ROX_VERSION` in the URL. Also, Golang's pattern is well-known in the K8S community.

Reply:
Is this what you mean?
https://github.com/stackrox/stackrox/pull/9417/commits/2c0cfa42b23208cac7695c13150b27478cace4af

Reply:
I meant using the text/template syntax. For example, you can replace `{{.Version}}` instead of `ROX_VERSION`.

If you actually want to use the `text/template` engine, it's fine, but I agree it might be overkill, and it would make more sense to use it for the whole config YAML.

Reply:
I see. How about this?
https://github.com/stackrox/stackrox/pull/9417/commits/0063fa51afe5a99eeef6c5fc5f8b1a86d9c87e21